### PR TITLE
test: 100% line coverage for TLG graph and listing functions

### LINKED ISSUE #1342: Enhancement: 100% test coverage for TLG functions and modules
### Description

The TLG section (graphs, listings, tables, and the Shiny module) needs full test coverage. This includes the plot functions in `R/` (`g_pkcg.R`, `t_pkct01.R`, `t_pkpt03.R`, etc.), the listing functions, and the Shiny module logic in `inst/shiny/modules/tab_tlg/`.

Currently some TLG functions have partial or no test coverage. Reaching 100% coverage will catch regressions as the TLG catalog expands and ensure output correctness.

### Definition of Done

- [ ] All TLG graph functions (`R/g_pkcg.R`, `R/g_lineplot.R`) have tests covering all scale types (LIN, LOG, SBS), summary methods, and edge cases
- [ ] All TLG table functions (`R/t_pkct01.R`, `R/t_pkpt03.R`, etc.) have tests covering grouping, formatting, and edge cases
- [ ] All TLG listing functions have tests
- [ ] Shiny module logic in `inst/shiny/modules/tab_tlg/` has tests where feasible (rendering, pagination, TLG dispatch)
- [ ] Test coverage report shows 100% line coverage for TLG-related files
- [ ] No regressions in existing tests

### PR DESCRIPTION
**Pull request by h5hoang**
## Issue

Closes #1342 

## Description

Adds unit tests to reach 100% `covr` line coverage on the TLG graph and listing files, and extends coverage of the TLG Shiny module where feasible.

```
(.venv) hoangh2@hoangh2-QW63QR aNCA % Rscript -e "
cov <- covr::package_coverage(path = '.', quiet = TRUE)
cat('\n--- Coverage for target files ---\n')
df <- as.data.frame(cov)
targets <- c('R/g_pkcg.R', 'R/g_lineplot.R', 'R/l_pkcl01.R')
for (f in targets) {
  rows <- df[df\$filename == f, ]
  pct  <- 100 * sum(rows\$value > 0) / nrow(rows)
  cat(sprintf('%-25s %.2f%%\n', f, pct))
}
"

Registered S3 method overwritten by 'tern':
  method   from 
  tidy.glm broom

--- Coverage for target files ---
R/g_pkcg.R                100.00%
R/g_lineplot.R            100.00%
R/l_pkcl01.R              100.00%
```

New test files:
- `test-g_pkcg_helpers.R` : internal helpers in `g_pkcg.R`: `generate_title`, `generate_subtitle`, `generate_title_mean`, `generate_subtitle_mean`, `keep_blq_timepoints`, `resolve_whiskers` (Both/Upper/Lower), `compute_summary_stats` (Mean_sdi/Mean_ci/Median_ci), `req_sbs_pkgs`
- `test-g_lineplot_helpers.R` : all 13 internal helpers in `g_lineplot.R` including `.build_plot_data` linetype branch and `.add_mean_layers`
- `test-tlg_option_modules.R` : `tlg_option_numeric`, `tlg_option_text`, `tlg_option_select` UI and server logic
- `test-tlg_option_table.R` : `tlg_option_table` UI and server logic (default rows, `$NA` conversion, confirm, cancel, add row, reset)

Modified test files:
- `test-tlg_module.R` : added `filter_tlg_excluded`, `.tlg_module_edit_widget` dispatch, and `tlg_module_server` pagination tests
- `test-l_pkcl01.R` : added mock test for the `requireNamespace` error path

## Definition of Done

- [x] All TLG graph functions (`g_pkcg.R`, `g_lineplot.R`) have tests covering all scale types, summary methods, and edge cases
- [x] All TLG table functions (`t_pkct01.R`, `t_pkpt03.R`) ; N/A (these files don't exist )
- [x] All TLG listing functions have tests (`l_pkcl01.R` at 100% coverage)
- [x] Shiny module logic in `inst/shiny/modules/tab_tlg/` has tests where feasible
- [x] `covr` reports 100% line coverage for `g_pkcg.R`, `g_lineplot.R`, and `l_pkcl01.R`
- [x] No regressions in existing tests

## How to test

Run the full test suite:
```r
devtools::test()
```

Verify coverage on the targeted files:

```
covr::file_coverage(
  source_files = c("R/g_pkcg.R", "R/g_lineplot.R", "R/l_pkcl01.R"),
  test_files   = list.files("tests/testthat", pattern = "^test-", full.names = TRUE)
)
```

## Contributor checklist
- [x] Code passes lintr checks
- [x] Code passes all unit tests
- [x] New logic covered by unit tests
- [x] New logic is documented
- [x] App or package changes are reflected in NEWS
- [x] Package version is incremented
- [ ] R script works with the new implementation (if applicable)
- [ ] Settings upload works with the new implementation (if applicable)
- [ ] If any `.scss` change was done, run `data-raw/compile_css.R`
- [ ] If a package dependency was added/changed, run `data-raw/test_suggests_hidden.R`

## Notes to reviewer
- The `tryCatch` error handler inside `tlg_module_server` (lines 181–190 of `tlg_module.R`) and the `remove_row` path in `tlg_option_table_server` are not covered by unit tests because both require a live browser session (`getReactableState` and `debounce` behavior differ in `testServer`). This is documented with comments in the respective test files.

### CODE REVIEW

File: tests/testthat/test-g_lineplot_helpers.R [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+    expect_true("facet_label" %in% names(result))
+    expect_true(any(grepl("Analyte1", result$facet_label)))
+    expect_true(any(grepl("n=1",      result$facet_label)))
+  })
+
+  it("appends dose-normalised suffix to color_var when linetype_by set", {
+    lt_data <- line_data
+    # Rows with non-empty linetype_by get a "(DN)" suffix on color_var;
+    # rows with empty string stay as-is.
+    lt_data$lt_var <- c("DN", "", "DN", "")
```
Comment (Gero1999) [inline tests/testthat/test-g_lineplot_helpers.R]: suggestion: I see you are repeating "DN" many times... If in the future we need to change the label it is always more comfortable to just have a variable storing the pattern so we only need to change one line to fix all tests at once

**Review by Gero1999 — APPROVED** [review]
Coverage is 100% for the relevant files and the lgoic of the tests seems good to me. Great job!

Image

I put some nitpicky comments but feel free to ignore

File: tests/testthat/test-tlg_module.R [hunk 2]
```diff
@@ -46,3 +56,136 @@ describe("filter_tlg_excluded", {
...
+# ---------------------------------------------------------------------------
+
+describe("tlg_module_server", {
+  test_data <- shiny::reactive(
+    list(conc = list(data = data.frame(
+      NFRLT = 1:3, AVAL = c(5, 4, 3), stringsAsFactors = FALSE
+    )))
+  )
+  render_list_ok  <- function(data, ...) list("plot_a", "plot_b", "plot_c")
+  render_list_err <- function(data, ...) stop("render failed")
```
Comment (Gero1999) [inline tests/testthat/test-tlg_module.R]: is this var used?
```suggestion
```

**Review by Gero1999 — APPROVED** [review]

**Comment by h5hoang** [comment]
Thanks for the review @Gero1999 , just addressed both comments!

**Review by Shaakon35 — APPROVED** [review]
good job @h5hoang  :) please merge. just update the description
